### PR TITLE
[12.0][IMP] fieldservice_maintenance: Convert Maintenance Equipment To FSM

### PR DESCRIPTION
**Pull request by osi-scampbell**
The following PR adds a wizard that allows users to convert a Maintenance Equipment into an FSM Equipment similar to how we convert Contacts into FSM Records

### CODE REVIEW

File: fieldservice_maintenance/wizard/maintenance_equipment_wizard.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# Copyright (C) 2018 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, models
+from odoo.exceptions import UserError
+
+
+class MainenanceEquipmentWizard(models.Model):
```
Comment (bodedra) [inline fieldservice_maintenance/wizard/maintenance_equipment_wizard.py]:
```suggestion
class MainenanceEquipmentWizard(models.TransientModel):
```

File: fieldservice_maintenance/__manifest__.py [hunk 2]
```diff
@@ -14,10 +14,12 @@
     ],
     'data': [
         'data/fsm_order_type.xml',
+        'security/ir.model.access.csv',
```
Comment (bodedra) [inline fieldservice_maintenance/__manifest__.py]:
```suggestion
```

**Review by bodedra — CHANGES_REQUESTED** [review]
Few points to improves:

- wizard should be `TransientModel`
- remove security file.
- create a lot number if product has tracking = serial or lot.

File: fieldservice_maintenance/wizard/maintenance_equipment_wizard.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    _name = 'maintenance.equipment.wizard'
+
+    @api.multi
+    def convert_maintenance_equipment_fsm(self):
+        maintenance_equpment_ids = self.env['maintenance.equipment'].\
+            browse(self._context.get('active_ids', []))
+        for maintenance_id in maintenance_equpment_ids:
+            if not maintenance_id.serial_no or not maintenance_id.serial_no:
```
Comment (bodedra) [inline fieldservice_maintenance/wizard/maintenance_equipment_wizard.py]: `maintenance_id.serial_no` condition appear twice in a condition.

File: fieldservice_maintenance/models/maintenance_equipment.py [hunk 4]
```diff
@@ -1,10 +1,11 @@
 # Copyright (C) 2018 Open Source Integrators
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
-from odoo import fields, models
+from odoo import api, fields, models
```
Comment (bodedra) [inline fieldservice_maintenance/models/maintenance_equipment.py:4]:
```suggestion
from odoo import fields, models
```

**Review by bodedra — CHANGES_REQUESTED** [review]
@osi-scampbell 

add this changes and squash your commits.

Once this is green, I will review it.

**Review by bodedra — CHANGES_REQUESTED** [review]
Functional review :+1:

**Comment by bodedra** [comment]
@brian10048 @hparfr @marcelsavegnago Are you guys available to review this?

**Review by bodedra — APPROVED** [review]

File: fieldservice_maintenance/wizard/maintenance_equipment_wizard.xml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+                target="new"
+                key2="client_action_multi"
+                groups="fieldservice.group_fsm_dispatcher"/>
+
+    <!-- wizard view -->
+    <record id="maintenance_fsm_wizard_view" model="ir.ui.view">
+        <field name="name">Convert to a FSM Equipment</field>
+        <field name="model">maintenance.equipment.wizard</field>
+        <field name="arch" type="xml">
+            <form string="Convert to a Field Service Worker or Location">
```
Comment (hparfr) [inline fieldservice_maintenance/wizard/maintenance_equipment_wizard.xml]: wrong string

**Review by hparfr — APPROVED** [review]
I'm not familiar with fsm_equipements so I don't know what are the benefits of the "convert to..."

If you want to align with fsm_location you will have to "inherits" https://github.com/OCA/field-service/blob/12.0/fieldservice/models/fsm_location.py#L10

Are you sure adding product_id should be part of this PR ?

**Review by marcelsavegnago — APPROVED** [review]

**Comment by osi-scampbell** [comment]
> I'm not familiar with fsm_equipements so I don't know what are the benefits of the "convert to..."
> 
> If you want to align with fsm_location you will have to "inherits" https://github.com/OCA/field-service/blob/12.0/fieldservice/models/fsm_location.py#L10
> 
> Are you sure adding product_id should be part of this PR ?

When an FSM Equipment is created, a Maintenance Equipment is automatically created. However when you create a Maintenance Equipment, an FSM Equipment is not automatically created. So this lets users take a maintenance equipment that has already been created and convert it to an FSM Equipment. 

You bring up a very good point about the product_id addition, I have had others ask the same question. I added this because on an FSM Equipment, product_id and lot_id are required fields. Perhaps this shouldn't be the case considering they aren't needed on Maintenance Equipment so maybe they shouldn't be needed on FSM Equipment? I will follow up on this question today and get back to you, I am not sure what the ramifications would be if we made product_id and lot_id no longer required on FSM Equipment

**Comment by dreispt** [comment]
https://travis-ci.com/github/OCA/field-service/jobs/450063553#L602

**Comment by osi-scampbell** [comment]
@dreispt It looks like these are lint errors in JavaScript that are in fieldservie and in fieldservice_vehicle_stock. Should those be fixed in this PR or should they be in a separate PR since they are in separate modules?

**Comment by marcelsavegnago** [comment]
ping @dreispt

**Pull request merged**